torchmultimodal/modules/layers/quantisation.py

```diff
@@ -66,33 +67,34 @@ def quantise(self, x_flat: Tensor):
         distances = torch.cdist(x_flat, self.embedding.weight, p=2.0) ** 2
 
         # Encoding - select closest embedding vectors
-        encoding_indices = torch.argmin(distances, dim=1)
+        codebook_indices = torch.argmin(distances, dim=1)
 
         # Quantise
-        quantised_flat = self.embedding(encoding_indices)
+        quantised_flat = self.embedding(codebook_indices)
 
         # Straight through estimator
         quantised_flat = x_flat + (quantised_flat - x_flat).detach()
 
-        return quantised_flat
+        return quantised_flat, codebook_indices
 
     def forward(self, x: Tensor):
         # Reshape and flatten encoder output for quantisation
         x_flat, permuted_shape = self._preprocess(x)
 
         # Quantisation via nearest neighbor lookup
-        quantised_flat = self.quantise(x_flat)
+        quantised_flat, codebook_indices = self.quantise(x_flat)
+        self.quantised_vectors = quantised_flat
+        self.codebook_indices = codebook_indices
 
         # Reshape back to original dims
         quantised = self._postprocess(quantised_flat, permuted_shape)
-        self.quantised_vectors = quantised
 
         return quantised
 
-    def get_quantised_vectors(self):
+    def get_quantised(self):
         # Retrieve the previously quantised vectors without forward passing again
-        if self.quantised_vectors is None:
-            raise Exception(
+        if self.quantised_vectors is None or self.codebook_indices is None:
+            raise NameError(
```

Comment:
why nameError here, whats the wrong name :) ?
assertion might make more sense

Reply:
internal linter seems to prefer the `if...raise` pattern over `assert` and doesn't like when I test a function for an assertion to be raised. The preferred pattern is if, raise specific error, see [earlier discussion](https://www.internalfb.com/diff/D36030926?dst_version_fbid=450544480168607&transaction_fbid=3255484671399666). 

`NameError` was the closest (raised when a local or global name is not found), but maybe I could change this to `AssertionError`.